HyperPlay-Gaming/hyperplay-desktop-client
# [FIX] Improve default Compatibility Layer setup

Closes https://github.com/HyperPlay-Gaming/hyperplay-desktop-client/pull/1045

Refactor the `getDefaultWine` method for better legibility and and fix not refreshing the wine list when called the first time.

### How to test (macOS and Linux)
- Make sure you dont have any downloaded wine or GPTK on your machine (delete the tools folder on hyperplay config folder on linux and mac)
- Start HyperPlay
- On Linux it should start downloading Wine-GE (can be seen on the logs or navigating quickly to Settings > Wine Manager)
- On MacOS it should download GPTK if you are on apple silicon and Sonoma or higher. Otherwise it will download Wine-Crossover.
- A notification should appear telling about the compatibility layer setup.
